### PR TITLE
Feature/proxy view container

### PR DESCRIPTION
**Pull request by vakrilov**

### CODE REVIEW

File: ui/core/view.ios.ts [hunk 1]
```diff
@@ -270,6 +270,29 @@ export class View extends viewCommon.View {
...
+    }
+
+    public _addToSuperview(superview: any, atIndex?: number): boolean {
+        if (superview && this._nativeView) {
+            var types = require("utils/types");
+
+            if (types.isNullOrUndefined(atIndex) || atIndex >= this._nativeView.subviews.count) {
+                superview.addSubview(this._nativeView);
+            }
+            else {
```
Comment (hdeshev) [inline ui/core/view.ios.ts]: Probably need to join the `else` with the line above.

File: ui/view-container/proxy-view-container.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
+﻿import types = require("utils/types");
+import view = require("ui/core/view");
+import definition = require("ui/proxy-view-container");
+import trace = require("trace");
+import layout = require("ui/layouts/layout-base");
+/**
+ * Proxy view container that adds all its native children dirctly to the parent. 
+ * To be used to be used as a logical grouping container of views.
```
Comment (hdeshev) [inline ui/view-container/proxy-view-container.ts]: "To be used" repeated twice.

File: ui/view-container/proxy-view-container.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ */
+// Cases to cover:
+// * Child is added to the attached proxy. Handled in _addViewToNativeVisualTree.
+// * Proxy (with children) is added to the DOM.
+//   - IOS: Handled in _addToSuperview - when the proxy is added, it adds all its children to the new parent.
+//   - Andoird: _onAttached calls _addViewToNativeVisualTree recoursively when the proxy is added to the parent.
```
Comment (hdeshev) [inline ui/view-container/proxy-view-container.ts]: Andoird -> Android

File: ui/view-container/proxy-view-container.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+// * Child is removed from attached proxy. Handled in _removeViewFromNativeVisualTree.
+// * Proxy (with children) is removed form the DOM.
+//   - IOS: Handled in _removeFromSuperview - when the proxy is removed, it removes all its children from its parent.
+//   - Andoird: _onDetached calls _removeViewFromNativeVisualTree recoursively when the proxy is removed from its parent.
```
Comment (hdeshev) [inline ui/view-container/proxy-view-container.ts]: Andoird -> Android

File: ui/view-container/proxy-view-container.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            let insideIndex = 0;
+            if (parent instanceof layout.LayoutBase) {
+                baseIndex = parent.getChildIndex(this);
+                baseIndex = parent._childIndexToNativeChildIndex(baseIndex);
+            }
+
+            if (types.isDefined(atIndex)) {
+                insideIndex = this._childIndexToNativeChildIndex(atIndex);
+            }
+            else {
```
Comment (hdeshev) [inline ui/view-container/proxy-view-container.ts]: Join the `else` with the line above?

File: ui/core/view-common.ts [hunk 6]
```diff
@@ -976,6 +976,27 @@ export class View extends proxy.ProxyObject implements definition.View {
         //
     }
 
+
+    _childIndexToNativeChildIndex(index?: number): number {
+        return index;
+    }
+
+    _getNativeViewsCount(): number {
```
Comment (hshristov) [inline ui/core/view-common.ts]: Why do we return 1 unconditionally?

File: ui/view-container/proxy-view-container.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                //add last;
+                insideIndex = this._getNativeViewsCount();
+            }
+            console.log("vc._addViewToNativeVisualTree at: " + atIndex + " base: " + baseIndex + " additional: " + insideIndex);
```
Comment (hdeshev) [inline ui/view-container/proxy-view-container.ts]: Maybe turn to trace or just delete the debug log.

File: ui/view-container/package.json [hunk 8]
```diff
@@ -0,0 +1,2 @@
+{ "name" : "proxy-view-container",
```
Comment (hdeshev) [inline ui/view-container/package.json]: We should probably change this to "view-container". Or let users import it as `ui/view-container/proxy-view-container`?

**Comment by dtopuzov** [comment]
run ci

File: ui/core/view.ios.ts [hunk 9]
```diff
@@ -265,6 +265,28 @@ export class View extends viewCommon.View {
         if (this._cachedFrame) {
             this._setNativeViewFrame(this._nativeView, this._cachedFrame);
         }
+    }
+
+    public _addToSuperview(superview: any, atIndex?: number): boolean {
+        if (superview && this._nativeView) {
+            var types = require("utils/types");
```
Comment (hshristov) [inline ui/core/view.ios.ts]: require it once and cache the result in order to get better performance.

File: ui/proxy-view-container/proxy-view-container.ts [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    public _addViewToNativeVisualTree(child: view.View, atIndex?: number): boolean {
+        trace.write("ViewContainer._addViewToNativeVisualTree for a child " + view + " ViewContainer.parent: " + this.parent, trace.categories.ViewHierarchy);
+        super._addViewToNativeVisualTree(child);
+
+        var parent = this.parent;
+        if (parent) {
+            let baseIndex = 0;
+            let insideIndex = 0;
+            if (parent instanceof layout.LayoutBase) {
+                baseIndex = parent.getChildIndex(this);
```
Comment (hshristov) [inline ui/proxy-view-container/proxy-view-container.ts]: One of the lines is not needed.

**Comment by hshristov** [comment]
:+1:

**Pull request merged**